### DISCUSSION TITLE
I need to insert a row in my database table for each user at their local midnight every day. What is the best way to store and handle time zones for each user to ensure the record is added at their local midnight?

### DISCUSSION THREAD

**Discussion by devanpellow** [supabase/supabase#29622] [Questions]
As the title suggests, I need to insert a row in my database table for each user at their local midnight every day. What is the best way to store and handle time zones for each user to ensure the record is added at their local midnight. 

I'm not sure how daylight savings would impact this sort of thing but any advice would really be appreciated 🙏

**Comment by bytaesu** [comment]
This supabase guide would be helpful.

> https://supabase.com/docs/guides/functions/schedule-functions

### 1. Save the time zone received from users in the user information db

- Create columns such as 'time_zone' and 'scheduled_at'.
- Save timezones in IANA Format (e.g. Asia/Seoul).
- Use something that can handle IANA time zone formats.
- If you are building a web project using Vercel, check out [vercel-example]( https://edge-functions-geolocation.vercel.sh). You can get user location.

### 2. Perform tasks every hour via pg_cron

1. Query users whose scheduled_at is earlier than the current time.
2. Call necessary Edge Functions for each user to perform database operations.
3. Update 'scheduled_at' to the next local midnight in UTC.

### 3. Consideration

Implement retry logic or a notification system for cases where Edge Functions fail.

**Reply by devanpellow** [reply]
Thank you! I'll take a look and try this out
